build kubernetes binaries failed

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
```
joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS=""
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS="-s -w"
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 
build/../build/common.sh: line 579: conditional binary operator expected
```
**What you expected to happen**:

build kubernetes binaries successfully

**How to reproduce it (as minimally and precisely as possible)**:
```
joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS=""
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS="-s -w"
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 
build/../build/common.sh: line 579: conditional binary operator expected
```

**Anything else we need to know?**:

**Environment**: 
- Kubernetes version (use `kubectl version`):  k8s v1.18.15+ , k8s v1.19.7 , k8s v1.20.2+
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):   my host environment: MacOS
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

@BenTheElder 

```
joshua:kubernetes joshua$ build/run.sh --version
build/../build/common.sh: line 579: conditional binary operator expected
joshua:kubernetes joshua$ 
joshua:kubernetes joshua$ 
joshua:kubernetes joshua$ make clean
build/../build/common.sh: line 579: conditional binary operator expected
make: *** [clean] Error 2
joshua:kubernetes joshua$ build/make-clean.sh 
build/../build/common.sh: line 579: conditional binary operator expected
joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS="" KUBE_VERBOSE=0
build/../build/common.sh: line 579: conditional binary operator expected
```

--- a/hack/lib/init.sh
+++ b/hack/lib/init.sh
@@ -50,6 +50,7 @@ source "${KUBE_ROOT}/hack/lib/util.sh"
 source "${KUBE_ROOT}/hack/lib/logging.sh"
 
 kube::log::install_errexit
+kube::util::ensure-bash-version
 
 source "${KUBE_ROOT}/hack/lib/version.sh"
 source "${KUBE_ROOT}/hack/lib/golang.sh"

--- a/hack/lib/util.sh
+++ b/hack/lib/util.sh
@@ -729,6 +729,17 @@ function kube::util::ensure_dockerized {
   fi
 }
 
+# kube::util::ensure-bash-version
+# Check if we are using a supported bash version
+#
+function kube::util::ensure-bash-version {
+  # shellcheck disable=SC2004
+  if ((${BASH_VERSINFO[0]}<5)); then
+    echo "ERROR: This script requires a minimum bash version of 5.0"
+    exit 1
+  fi
+}
+
 # kube::util::ensure-gnu-sed
 # Determines which sed binary is gnu-sed on linux/darwin
 #

